src/dali/abstract_pair_converter_plugin.py

```diff
@@ -12,25 +12,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from datetime import datetime
-from typing import Dict, NamedTuple, Optional, cast
+from datetime import datetime, timedelta
+from json import JSONDecodeError
+from typing import Any, Dict, List, NamedTuple, Optional, cast
 
-from rp2.rp2_decimal import RP2Decimal
+import requests
+from requests.models import Response
+from requests.sessions import Session
+from rp2.rp2_decimal import ZERO, RP2Decimal
 from rp2.rp2_error import RP2TypeError
 
 from dali.cache import load_from_cache, save_to_cache
 from dali.configuration import HISTORICAL_PRICE_KEYWORD_SET
 from dali.historical_bar import HistoricalBar
 from dali.logger import LOGGER
 
+# exchangerates.host keywords
+_SUCCESS: str = "success"
+_SYMBOLS: str = "symbols"
+_RATES: str = "rates"
+
+# exchangerates.host urls
+_EXCHANGE_BASE: str = "https://api.exchangerate.host/"
+_EXCHANGE_SYMBOLS: str = "https://api.exchangerate.host/symbols"
+
+_DAYS_IN_SECONDS: int = 86400
+_FIAT_EXCHANGE: str = "Exchangerate.host"
+
+# First on the list has the most priority
+_FIAT_PRIORITY: List[str] = ["USD", "JPY", "EUR", "GBP", "AUD"]
```

Comment:
I'm a bit concerned this order is a little arbitrary (if not, can you add a link to the source?) and it may change in the future (so it requires maintenance): this is fine for the first version, but can you add a comment saying that in the future we should explore the idea of creating this list dynamically (either at runtime or perhaps have a script that we run every once in a while and that generates the hardcoded list)?

Reply:
It's based on [markets sorted by volume for BTC](https://coinmarketcap.com/currencies/bitcoin/markets/). I accidentally left out KRW though, it should be after JPY.

I'm not sure how this list could be reliably created dynamically. We could require users to get a free coinmarketcap API account and pull the data for all the markets for BTC, add all duplicate fiat markets' volumes together and then sort the list. I see that being the most reliable in the short term.

Do you have any ideas?

Reply:
Thinking on this a little more. I would think that this priority should actually be able to be user-defined. Since if a user was actually in Nigeria and needed NGN denominated amounts this default priority wouldn't work. 

Even for me, I would want JPY preferred over USD when calculating JPY taxes.

Reply:
I like the idea of it being user defined: let's have a reasonable default and let users override it if they want. It can be an optional argument to the plugin constructor.

Reply:
OK, I added it as a user defined list in the config file and then parsed with json:
```
from json import loads

def __init__(self, historical_price_type: str, fiat_priority: Optional[str] = None) -> None:
        ...
        if priority_fiat is None:
            self.__fiat_priority = _FIAT_PRIORITY
        else:
            self.__fiat_priority = loads(fiat_priority) 
```